ibmdbpy/geoFrame.py

```diff
@@ -1028,22 +1028,9 @@ def _binary_operation_handler(self, ida2, db2gse_function,
 
         # Get the definitions of the columns, which will be the arguments for
         # the DB2GSE function
-        column1_for_db2gse = ida1.internal_state.columndict[ida1.geometry.column] 
-        if column1_for_db2gse[0] == '\"' and column1_for_db2gse[-1] == '\"':
-            column1_for_db2gse = column1_for_db2gse[1:-1]
-        column2_for_db2gse = ida2.internal_state.columndict[ida2.geometry.column] 
-        if column2_for_db2gse[0] == '\"' and column2_for_db2gse[-1] == '\"':
-            column2_for_db2gse = column2_for_db2gse[1:-1]
-
-        arguments_for_db2gse_function = []
-        if 'IDA1' or 'DB2GSE' in column1_for_db2gse:
-            arguments_for_db2gse_function.append(column1_for_db2gse) 
-        else:
-            arguments_for_db2gse_function.append('IDA1.'+column1_for_db2gse)
-        if 'IDA2' or 'DB2GSE' in column2_for_db2gse:
-            arguments_for_db2gse_function.append(column2_for_db2gse)
-        else:
-            arguments_for_db2gse_function.append('IDA2.'+column2_for_db2gse)
+        column1_for_db2gse = 'IDA1.\"%s\"' %(ida1.geometry.column)
```

Comment:
Please clarify this change. This is now basically the code from before eva's changes in https://github.com/ibmdbanalytics/ibmdbpy/pull/50/files#diff-5624c304e87b1a80d0a03491cd424a55, which, as I understand, did not work correctly in all cases. I admit that that's far better than not working in any case, which we have now, but she was trying to solve a problem with that change.

Reply:
The main difference is that before the change the value of column1_for_db2gse (and column2_for_db2gse)  can be an expression like 'DB2GSE.ST_MBR("SHAPE")'. After the change it is the name of a column, the geometry column, like 'MBR'. This suffices because the association between the defining expression and the column name is included in the SQL statement of the name attribute, for instance:
``
'(SELECT * FROM (SELECT "OBJECTID","SHAPE","STATEFP","COUNTYFP","COUNTYNS","NAME","GEOID","NAMELSAD","LSAD","CLASSFP","MTFCC","CSAFP","CBSAFP","METDIVFP","FUNCSTAT","ALAND","AWATER","INTPTLAT","INTPTLON",DB2GSE.ST_MBR("SHAPE") AS "MBR" FROM SAMPLES.GEO_COUNTY) WHERE ("NAME" = \'Austin\'))'
``

Reply:
It's still not entirely clear to me how this works, but it's better than the current situation :-D